tasks/service-runner/app/service/service-metrics.ts

```diff
@@ -0,0 +1,70 @@
+import { Response, Request, NextFunction } from 'express';
+import axios from 'axios';
+import Agent from 'agentkeepalive';
+import env from '../util/env';
+
+/**
+ * Get prometheus-compatible metric message from harmony backend
+ * @param serviceID - The service name (essentially the image name)
+ * @returns Promise of prometheus-compatible metric message
+ */
+async function _getHarmonyMetric(serviceID: string): Promise<string> {
+
+  const timeout = 3_000; // Wait up to 3 seconds for the server to start sending
+  const activeSocketKeepAlive = 6_000;
+  const maxSockets = 1;
+  const maxFreeSockets = 1;
+  const keepaliveAgent = new Agent({
```

Comment:
Yes, but as I mentioned above:
```
There were very nice suggestions during tech meeting such as re-using the axios call from pull_worker, but I decided not doing that since they have different validateStatus.
```

Reply:
I'm not following the `validateStatus` part.

The thing to reuse is the the `keepAliveAgent`. So instead of defining a new one on lines 17-23 you would use the same agent as the pull worker. The only change to the call to `axios.get` would be to reuse the `keepAliveAgent` instead of a new one.

Reply:
Ah, I missed that ... let me change it.

Reply:
Done and pushed ... thanks!